pkg/controller/volume/persistentvolume/provision_test.go

```diff
@@ -24,9 +24,15 @@ import (
 	storage "k8s.io/api/storage/v1"
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	api "k8s.io/kubernetes/pkg/apis/core"
 )
 
+func init() {
+	// TODO (#71091) modify these tests to work with the feature gate on OR modify code to tolerate nil StorageClass.VolumeBindingMode
```

Comment:
This looks unrelated but without this hack, the tests will fail because VolumeScheduling gets turned on somehow. It should be on to begin with so I don't know if the constant feature gate toggling is causing it or what. Either way, we will need to either a) add a non-nil VolumeBindingMode to every StorageClass, just as I have added a non-nil VolumeMode to every PV/PVC or b) consider again allowing VolumeBindingMode and VolumeMode to be nil to make writing tests less unpleasant. ref: https://github.com/kubernetes/kubernetes/issues/71091

Reply:
It is indeed very odd that turning on/off the block feature gate would impact the volumescheduling feature gate :-/

Reply:
I think it's because simply importing pkg/features/kube_features.go inits the default feature gates

Reply:
we cannot flip feature gates in init blocks like this. that inverts defaults for the rest of the unit test run.

Reply:
having to toggle a feature that is on by default off to get the tests to pass seems bad. can you describe why we're doing this? is it just to avoid sweeping fixtures, or is there a bug in how we handle these objects?

Reply:
@liggitt the VolumeScheduling feature requires all storageClasses to have a VolumeBindingMode but the ones created in these tests don't. These tests passed before because the feature was left "off" by a previous test (probably https://github.com/kubernetes/kubernetes/pull/65829/commits/f23b41c4bc543a80a2a3d6ab6ad0bd9115d570e0#diff-f7b8e9bcc0deeca2eea2e5331bd2e8c2R343) and I think by importing the features package here I've toggled VolumeScheduling back "on," causing these tests to fail. I don't think this PR is the place to address it in full so I created https://github.com/kubernetes/kubernetes/issues/71091 to fix it later with these TODOs referencing it

Reply:
basically somebody else is doing what you told me not to, causing issues for me here, "we cannot flip feature gates in init blocks like this. that inverts defaults for the rest of the unit test run" :p and we should fix it in a separate PR imo

Reply:
Either me or @lichuqiang will fix the VolumeScheduling feature gate toggling in these tests, but let's not block the raw block feature on this.

Reply:
> let's not block the raw block feature on this

unfortunately, I disagree. with flipped flags leaking from other packages, I have little confidence in our tests. I'm working on adding checks to make sure no changes escape a package test, will work with @msau42 to fix leaking tests, then return to this once that is merged.